### PR TITLE
Added planner specific parameters to ompl_planning.yaml emitter.

### PR DESCRIPTION
**Pull request by drchrislewis**
Each parameter is set to current defaults. This is fragile, as defaults may change.
However, yaml makes it difficult to provide correct syntax for parameters as comments which would have been preferable. 

Each parameter also has comments which help users understand what it means.

### CODE REVIEW

File: include/moveit/setup_assistant/tools/moveit_config_data.h [hunk 1]
```diff
@@ -76,6 +76,47 @@ struct GroupMetaData
   int kinematics_solver_attempts_; // solver attempts
 };
 
+/**
+ * Planning parameters which may be set in the config files
+ */
+ struct OmplPlanningParameter{
```
Comment (isucan) [inline include/moveit/setup_assistant/tools/moveit_config_data.h]: This is not so important, but i like spaces before {

File: include/moveit/setup_assistant/tools/moveit_config_data.h [hunk 2]
```diff
@@ -76,6 +76,47 @@ struct GroupMetaData
...
+ };
+
+/** \brief This class describes the OMPL planners by name, type, and parameter list, used to create the ompl_planning.yaml file */
+ class OMPLPlannerDescription{
+ public:
+/** \brief Constructor
+ *  @param name: name of planner
+ *  @parameter type: type of planner
+ */
+   OMPLPlannerDescription(const std::string  & name, const std::string  & type){
```
Comment (isucan) [inline include/moveit/setup_assistant/tools/moveit_config_data.h]: And we use & next to the variable name in MoveIt.

File: src/tools/moveit_config_data.cpp [hunk 3]
```diff
@@ -296,18 +296,87 @@ bool MoveItConfigData::outputOMPLPlanningYAML( const std::string& file_path )
...
   emitter << YAML::Value << YAML::BeginMap;
 
-  // Add Planners
-  static const std::string planners[] =
-    {"SBL", "EST", "LBKPIECE", "BKPIECE", "KPIECE", "RRT", "RRTConnect", "RRTstar", "TRRT", "PRM", "PRMstar" };
-
+  std::vector<OMPLPlannerDescription> planner_des;
+
+  OMPLPlannerDescription SBL("SBL","geometric");
+  SBL.addParameter("range","0.0","Max motion added to tree. ==> maxDistance_ default: 0.0, if 0.0, set on setup()");
```
Comment (isucan) [inline src/tools/moveit_config_data.cpp]: Please add a space after every ,

**Comment by isucan** [comment]
+1, assuming the build error is fixed.

File: include/moveit/setup_assistant/tools/moveit_config_data.h (hunk 1 above)
Comment (drchrislewis) [inline include/moveit/setup_assistant/tools/moveit_config_data.h]: I inserted either a space or a carriage return before {

FYI I plan to respond individually to each email notice I get. I'm fixing
all your request at once. In the future, I'd be happy to respond to blocks
of request with a single email if you'd like that better.

File: include/moveit/setup_assistant/tools/moveit_config_data.h (hunk 2 above)
Comment (drchrislewis) [inline include/moveit/setup_assistant/tools/moveit_config_data.h]: fixed. It does look better this way.

File: src/tools/moveit_config_data.cpp (hunk 3 above)
Comment (drchrislewis) [inline src/tools/moveit_config_data.cpp]: fixed. I usually do this, not sure why I slipped up here.

**Comment by drchrislewis** [comment]
please explain both the +1, and the build error.
My intuition says +1 is same as kudos.

It seems to build, run, and the resulting file works with the GUI.
The main issue I have is making sure the parameter names an default values
I provide are identical to those in the source of ompl.
I tried not to change the default behavior. Toward this goal I would have
liked to have generated commented parameters like:

# range: 0.0  #  Maximum distance from current graph to grow ....

However, yaml's emitter did not want to do that... at all.

**Comment by isucan** [comment]
+1 means that i agree this should be merged.
the Travis build that run for your previous version had failed (I did not check why), but it passes now. I will let someone else complete the merge, as this is a config change, and more eyes are better :)

**Comment by davetcoleman** [comment]
Looks good

**Pull request merged**